### DISCUSSION TITLE
There is an issue with the IF activity in Elsa 3.1.3 designer

### DISCUSSION THREAD

**Discussion by shxylgithub** [elsa-workflows/elsa-core#5684] [General]
Hello, teacher

In Elsa 3.1.3 designer, if an IF activity displays two buttons in the activity display, Then and Else do not work. If you click the "Then" or "Else" button, it is not possible to logically drag and drop a new activity into the sub flowchart? Is it not possible to visualize IF activities in Elsa 3.1.3 designer?
![If](https://github.com/elsa-workflows/elsa-core/assets/171897414/661c8306-4f2b-41c8-bcd1-1e9212859296)

**Comment by glime-ai** [comment]
Based on the context provided, it appears that you are experiencing an issue with the IF activity in the Elsa 3.1.3 designer where the Then and Else branches are not functioning as expected. This could be a known limitation or a bug in the Elsa Workflow designer.

As of the information available, the Elsa Workflow designer has some known limitations, such as the designer not yet fully supporting embedding in other applications and certain activities not being fully tested. However, there is no specific mention of the IF activity not functioning correctly in the designer in the provided context.

To address your issue, I would recommend checking the Elsa GitHub repository for any reported issues related to the IF activity in the designer. If you do not find a relevant issue, you may want to report this as a new issue to get assistance from the Elsa community and developers. They might provide a workaround or inform you about a potential fix in a future release.

In the meantime, you can define your workflows programmatically as a workaround until the designer support for the IF activity is resolved.

If you need further assistance or have more questions, you might want to reach out to the Elsa community or check for updates on the GitHub repository.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
